Add interface level object definitions for BeaconBlock endpoint.

rolfyone opened the pull request:
Added tests for some utility functions, moved functional tests from the interface level down to the ChainDataProvider.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

#1301

rolfyone committed:
Add interface level object definitions for BeaconBlock endpoint.

Added tests for some utility functions, moved functional tests from the interface level down to the ChainDataProvider.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

#1301

rolfyone committed:
rename RestApiUtils to be more specific

Start a CacheControlUtils

Refactor the context handling in BeaconBlockHandler.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
Merge remote-tracking branch 'upstream/master' into 1301-beaconBlockHandler

rolfyone committed:
Merge remote-tracking branch 'upstream/master' into 1301-beaconBlockHandler

macfarla commented on data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/CacheControlUtilsTest.java:
```diff
@@ -0,0 +1,64 @@
+  @Mock private ChainDataProvider provider;
+
+  @Test
+  void getMaxAgeForSignedBlock_shouldSetZeroIfNotFinalized() {
```
```suggestion
  void getMaxAgeForSignedBlock_shouldReturnZeroIfNotFinalized() {
```

macfarla commented on data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/CacheControlUtilsTest.java:
```diff
@@ -0,0 +1,64 @@
+  }
+
+  @Test
+  void getMaxAgeForSignedBlock_shouldSetZeroIfFinalized() {
```
```suggestion
  void getMaxAgeForSignedBlock_shouldReturnZeroIfFinalized() {
```

macfarla replied:
hang on they aren't both zero are they?

rolfyone replied:
ok, bad naming of the test.
If the object is not finalized, i want to set max-age=zero in the cache header, otherwise setting to the cache-finalized string (cache timeout = 1 year)
Hopefully the new names make more sense..

rolfyone committed:
Update data/beaconrestapi/src/test/java/tech/pegasys/artemis/beaconrestapi/CacheControlUtilsTest.java

Co-Authored-By: Sally MacFarlane <sally.macfarlane@consensys.net>

macfarla commented on data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSPubKey.java:
```diff
@@ -0,0 +1,66 @@
+
+  public BLSPubKey(Bytes bytes) {
+    checkArgument(
+        bytes.size() == SIZE, "Bytes4 should be %s bytes, but was %s bytes.", SIZE, bytes.size());
```
why is Bytes4 48 bytes?

rolfyone replied:
comment should be bringing in size, i'll fix those.

macfarla commented on data/provider/src/main/java/tech/pegasys/artemis/api/schema/BLSSignature.java:
```diff
@@ -0,0 +1,66 @@
+
+  public BLSSignature(Bytes bytes) {
+    checkArgument(
+        bytes.size() == SIZE, "Bytes4 should be %s bytes, but was %s bytes.", SIZE, bytes.size());
```
should it just be Bytes should be ...

rolfyone committed:
changes to hopefully clear up test naming, from review feedback

Signed-off-by: Paul Harris <paul.harris@consensys.net>

macfarla commented on data/provider/src/main/java/tech/pegasys/artemis/provider/BLSSignatureSerializer.java:
```diff
@@ -17,12 +17,12 @@
   public void serialize(BLSSignature value, JsonGenerator gen, SerializerProvider serializers)
       throws IOException {
-    gen.writeString(value.toBytes().toHexString().toLowerCase());
+    gen.writeString(value.toHexString().toLowerCase());
```
is this change related?

rolfyone replied:
because BLSSignature changed, and the toBytes isn't on the new object... We're the only consumers of the serializers, so i updated it.

macfarla commented on data/provider/src/main/java/tech/pegasys/artemis/provider/JsonProvider.java:
```diff
@@ -38,6 +39,7 @@ private void addTekuMappers() {
 
     module.addDeserializer(BLSPublicKey.class, new BLSPublicKeyDeserializer());
     module.addSerializer(BLSPublicKey.class, new BLSPublicKeySerializer());
+    module.addSerializer(BLSPubKey.class, new BLSPubKeySerializer());
```
why do we have BLSPublicKey and BLSPubKey?

rolfyone replied:
as discussed, will stick with spec name, and remove the serializer for BLSPublicKey

macfarla commented on data/provider/src/test/java/tech/pegasys/artemis/api/ChainDataProviderTest.java:
```diff
@@ -51,9 +53,9 @@
-  private static UnsignedLong epoch;
+  private final tech.pegasys.artemis.datastructures.blocks.SignedBeaconBlock signedBeaconBlock =
+      DataStructureUtil.randomSignedBeaconBlock(1, 1);
+  //  private static UnsignedLong epoch;
```
stray comment - is epoch not needed?

macfarla commented on data/provider/src/test/java/tech/pegasys/artemis/api/ChainDataProviderTest.java:
```diff
@@ -174,4 +176,86 @@ public void getGenesisTime_shouldReturnValueIfStoreAvailable() {
+  @Test
+  public void getBlockByBlockRoot_shouldReturnEmptyWhenStoreNotFound()
+      throws ExecutionException, InterruptedException {
+    CombinedChainDataClient myClient = mock(CombinedChainDataClient.class);
```
myClient??

rolfyone replied:
renamed

macfarla commented on data/beaconrestapi/src/main/java/tech/pegasys/artemis/beaconrestapi/beaconhandlers/BeaconBlockHandler.java:
```diff
@@ -66,65 +77,61 @@ public BeaconBlockHandler(
-        final Bytes32 blockParam =
-            Bytes32.fromHexString(validateQueryParameter(queryParamMap, ROOT));
+      if (queryParamMap.containsKey(ROOT)) {
+        final Bytes32 blockParam = getParameterValueAsBytes32(queryParamMap, ROOT);
```
```suggestion
        final Bytes32 blockRoot = getParameterValueAsBytes32(queryParamMap, ROOT);
```

rolfyone replied:
will do this as part of bigger commit.

macfarla reviewed (commented):
looks good overall. some comments

rolfyone committed:
changes per review comments.

Can't remove BLSPublicKeySerializer yet, because its used in beacon/state, but i'll clean that up when i do that endpoint.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

rolfyone committed:
Merge branch 'master' into 1301-beaconBlockHandler
